### ISSUE TITLE
Train loss averaged over nominal `len(train_loader)` instead of batches actually seen

### ISSUE THREAD

**Issue by timoverse** [pwhofman/probly#558]
`_training_loop` and `_training_loop_relative_likelihood` (`src/probly_benchmark/train.py`) 
accumulate the per-batch loss over the *actual* batches yielded by the loader but divide by
`len(train_loader)`, the nominal length. On ImageNet webdataset loaders the actual
batch count exceeds the nominal one (partial final batch per worker), so the logged
`train_loss` (and `probe/`, `member_i/`, `phase1/` variants) is systematically
inflated by roughly the ratio of actual/nominal batches.

**Affected:** logged training losses of all methods on ImageNet. Metric-only — no
effect on optimization, model selection (which uses validation loss), or test
metrics. CIFAR-10 unaffected.

**Proposed fix**: count batches in the loop and
divide by `max(num_batches, 1)`:

```python
running_loss = 0.0
num_batches = 0
for inputs_, targets_ in tqdm(train_loader):
    ...
    num_batches += 1
running_loss /= max(num_batches, 1)
```

Note the same nominal-vs-actual mismatch also exists in the `validate_*` functions
(`val_loss / len(val_loader)` in `src/probly_benchmark/train_funcs.py`); fix
together.
